src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteAndCreateOrgMetaEntitySets.kt

```diff
@@ -0,0 +1,113 @@
+            logger.info("finished initializing org metadata entity sets - ${it.title} (${it.id})")
+
+            logger.info("syncing org external database tables and columns - ${it.title} (${it.id})")
+            syncOrgExternalDatabaseObjects(it)
+            logger.info("finished syncing org external database tables and columns - ${it.title} (${it.id})")
+
+            logger.info("finished processing org - ${it.title} (${it.id})")
+            logger.info("================================")
+            logger.info("================================")
+        }
+
+        return true
+    }
+
+    private fun deleteOrgMetaEntitySet(org: Organization, entitySetId: UUID) {
+        if (entitySetId != UNINITIALIZED_METADATA_ENTITY_SET_ID) {
+            logger.info("getting org metadata entity set (${entitySetId}) - ${org.title} (${org.id})")
+            val entitySet = entitySetsService.getEntitySet(entitySetId)
+            if (entitySet != null) {
+                logger.info("deleting org metadata entity set (${entitySetId}) - ${org.title} (${org.id})")
+                entitySetsService.deleteEntitySet(entitySet)
+            }
+            else {
+                logger.info("org metadata entity set (${entitySetId}) is null - ${org.title} (${org.id})")
+            }
+        }
+        else {
+            logger.info("org metadata entity set (${entitySetId}) is UNINITIALIZED - ${org.title} (${org.id})")
+        }
+    }
+
+    private fun syncOrgExternalDatabaseObjects(org: Organization) {
+
+        logger.info("getting org external database tables - ${org.title} (${org.id})")
+        val tables = externalDatabaseManagementService.getExternalDatabaseTables(org.id)
+        logger.info("table ids - ${tables.keys}")
+
+        logger.info("getting org external database columns - ${org.title} (${org.id})")
+        val columns = tables
+            .keys
+            .map { id -> externalDatabaseManagementService.getExternalDatabaseTableWithColumns(id) }
+            .associateBy(
+                { pair -> pair.table.id },
+                { pair -> pair.columns }
+            )
+        val columnIds = columns.mapValues { it.value.map { c -> c.id }.toSet() }
+        logger.info("column ids by table id - $columnIds")
+
+        logger.info("adding org external database tables and columns")
+        metadataEntitySetsService.addDatasetsAndColumns(org.id, tables.values, columns)
+
+        tables.keys.forEach { id ->
+            val key = AclKey(id)
+            var ids = auditRecordEntitySetsManager.getAuditRecordEntitySets(key)
+            var entitySets = entitySetsService.getEntitySetsAsMap(ids)
+            var propertyTypes = entitySetsService.getPropertyTypesOfEntitySets(ids).mapValues { it.value.values }
+            logger.info("adding audit entity sets associated with org external database tables")
+            metadataEntitySetsService.addDatasetsAndColumns(entitySets.values, propertyTypes)
```

Comment:
Do we actually want to store audit entity set info? We're not currently storing anything in audit entity sets, they make the UI super cluttery and ugly (and misleading, given that they're empty), and it would be really easy to add them in a future migration if/when we actually start using them (although I really very much think they should not be surfaced through the same workflow as viewing a regular dataset object because that will be forever confusing, but that's a discussion for the future)

Reply:
It's a good question, not sure what the answer is. I added this in because it's a bug I found in that search was returning audit entity sets before but not after running the migration. 

That being said, the metadata entity now has `ol.flags` which we can use to filter out audit entity sets.

Reply:
Yeah, I also realize the code will initialize any new audit entity sets that get created so we may as well go ahead and do this here, although I do think we should definitely update searches in the frontend to require AUDIT to not be present in ol.flags since it has made the experience on staging orgs app / local very painful even with context on audit entity sets

Reply:
Agreed. I have a ticket for it, just blocked by this release.

https://jira.openlattice.com/browse/APPS-2540